pandas/core/dtypes/cast.py

```diff
@@ -2017,6 +2019,18 @@ def np_can_hold_element(dtype: np.dtype, element: Any) -> Any:
                 if element._hasna:
                     raise ValueError
                 return element
+            elif tipo.itemsize > dtype.itemsize:
+                if isinstance(element, np.ndarray):
+                    # e.g. TestDataFrameIndexingWhere::test_where_alignment
+                    casted = element.astype(dtype)
```

Comment:
can we just do this in array_equivalent?

Reply:
sure.  will change in follow-up, as CI is a PITA these days

Reply:
updated + greenish